File: src/main/AnswerOption.java
package main;

public class AnswerOption {
    private int id;
    private Question question;
    private String content;
    private int numberCorrectAnswer;
    private String answerType;

    public AnswerOption(int id, String content) {
        this.id = id;
        this.content = content;
    }

    public AnswerOption(String content, int correctAnswer) {
        this.content = content;
        this.numberCorrectAnswer = correctAnswer;
    }

    public AnswerOption(int id, String content, int correctAnswer) {
        this.id = id;
        this.content = content;
        this.numberCorrectAnswer = correctAnswer;
    }

    public AnswerOption(int id, String content, int correctAnswer, String answerType) {
        this.id = id;
        this.content = content;
        this.numberCorrectAnswer = correctAnswer;
        this.answerType = answerType;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Question getQuestion() {
        return question;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public int getNumberCorrectAnswer() {
        return numberCorrectAnswer;
    }

    public void setNumberCorrectAnswer(int correctAnswer) {
        this.numberCorrectAnswer = correctAnswer;
    }

    public String getAnswerType() {
        return answerType;
    }

    public void setAnswerType(String answerType) {
        this.answerType = answerType;
    }


    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + id;
        result = prime * result + ((content == null) ? 0 : content.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        AnswerOption other = (AnswerOption) obj;
        if (id != other.id)
            return false;
        if (content == null) {
            if (other.content != null)
                return false;
        } else if (!content.equals(other.content))
            return false;
        return true;
    }

    @Override
    public String toString() {
        return "AnswerOption [id=" + id + ", content=" + content + "]";
    }

}

File: src/QuizAppCSharp/QuizApp/Model/AnswerOption.cs
using System;
using System.Collections.Generic;

namespace QuizApp
{
    public class AnswerOption
    {
        public AnswerOption(int id, string content, int numberCorrectAnswer)
        {
            Id = id;
            Content = content;
            NumberCorrectAnswer = numberCorrectAnswer;
        }
        public AnswerOption(int id, string content)
        {
            Id = id;
            Content = content;
        }

        public int Id {  get; set; }
        public Question Question { get; set; }
        public string Content { get; set; }
        public int NumberCorrectAnswer { get; set; }
        public string AnswerType { get; set; }

        public override bool Equals(object obj)
        {
            return obj is AnswerOption option &&
                   Id == option.Id &&
                   Content == option.Content;
        }

        public override int GetHashCode()
        {
            int hashCode = 1058685026;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Content);
            return hashCode;
        }

    }
}